Apps/Src/UpdateDisplay.c

```diff
@@ -0,0 +1,335 @@
+static UpdateDisplayError_e UpdateDisplay_PutNext(DisplayCmd_t cmd){
+	CPU_TS ticks;
+	OS_ERR err;
+
+	OSMutexPend(&DisplayQ_Mutex, 0, OS_OPT_PEND_BLOCKING, &ticks, &err);
+  assertOSError(OS_DISPLAY_LOC, err);
+	
+	bool success = disp_fifo_put(&msg_queue, cmd);
+
+	OSMutexPost(&DisplayQ_Mutex, OS_OPT_POST_ALL, &err);
+	assertOSError(OS_DISPLAY_LOC, err);
+
+	if(success){
+		OSSemPost(&DisplayQ_Sem4, OS_OPT_POST_ALL, &err);
+		assertOSError(OS_DISPLAY_LOC, err);
+	}
+	else{
+		assertUpdateDisplayError(UPDATEDISPLAY_ERR_FIFO_PUT);
+		return UPDATEDISPLAY_ERR_FIFO_PUT;
+	}
+
+	return UPDATEDISPLAY_ERR_NONE;
+}
+
+/**
+ * @brief Several elements on the display do not update their
+ * state until a touch/click event is triggered. This includes the
+ * blinkers, gear selector, cruise control and regen braking indicator.
+ * @returns UpdateDisplayError_e
+ */
+static UpdateDisplayError_e UpdateDisplay_Refresh(){
+	DisplayCmd_t refreshCmd = {
+		.compOrCmd = "click",
+		.attr = NULL,
+		.op = NULL,
+		.numArgs = 2,
+		.argTypes = {INT_ARG,INT_ARG},
+		{
+			{.num=0},
+			{.num=1}
+		}
+	};
+
+	UpdateDisplayError_e ret = UpdateDisplay_PutNext(refreshCmd);
+	assertUpdateDisplayError(ret);
+	return ret;
+}
+
+/**
+ * @brief Uses component enum to make assigning component values easier.
+ * Differentiates between timers, variables, and components to assign values.
+ * @param comp component to set value of
+ * @param val value
+ * @return UpdateDisplayError_e
+ */
+static UpdateDisplayError_e UpdateDisplay_SetComponent(Component_e comp, uint32_t val){
+	UpdateDisplayError_e ret = UPDATEDISPLAY_ERR_NONE;
+	
+	// For components that are on/off
+	if(comp <= MOTOR && val <= 1){
+		DisplayCmd_t visCmd = {
+			.compOrCmd = "vis",
+			.attr = NULL,
+			.op = NULL,
+			.numArgs = 2,
+			.argTypes = {STR_ARG,INT_ARG},
+			{
+				{.str=(char*)compStrings[comp]},
+				{.num=val}
+			}
+		};
+		
+		ret = UpdateDisplay_PutNext(visCmd);
+		assertUpdateDisplayError(ret);
+		return ret;
+	}
+	// For components that have a non-boolean value
+	else if(comp > MOTOR){
+		DisplayCmd_t setCmd = {
+			.compOrCmd = (char*)compStrings[comp],
+			.attr = "val",
+			.op = "=",
+			.numArgs = 1,
+			.argTypes = {INT_ARG},
+			{
+				{.num=val}
+			}
+		};
+
+		ret = UpdateDisplay_PutNext(setCmd);
+		assertUpdateDisplayError(ret);
+		return UpdateDisplay_PutNext(setCmd);
+	}
+	else{
+		assertUpdateDisplayError(UPDATEDISPLAY_ERR_PARSE_COMP);
+		return UPDATEDISPLAY_ERR_PARSE_COMP;
+	}
+	return UPDATEDISPLAY_ERR_NONE;
+}
+
+UpdateDisplayError_e UpdateDisplay_SetPage(Page_e page){
+	DisplayCmd_t pgCmd = {
+		.compOrCmd = "page",
+		.attr = NULL,
+		.op = NULL,
+		.numArgs = 1,
+		.argTypes = {INT_ARG},
+		{
+			{.num=page}
+		}
+	};
+
+	UpdateDisplayError_e ret = UpdateDisplay_PutNext(pgCmd);
+	return ret;
+}
+
+/* WRAPPERS */
+UpdateDisplayError_e UpdateDisplay_SetSOC(uint8_t percent){	// Integer percentage from 0-100
+	UpdateDisplayError_e ret = UpdateDisplay_SetComponent(SOC, percent);
```

Comment:
maybe add a static variable so that you ensure you're not constantly setting the value over and over again when it hasn't changed.

Reply:
Same applies for the rest of the functions that come after this.

Reply:
Not a critical change though, just a suggestion.